app/models/test.rb

```diff
@@ -1,2 +1,8 @@
 class Test < ApplicationRecord
+  def self.by_category(name)
+    joins('JOIN categories on categories.id = tests.category_id')
+      .where('categories.title = ?', name)
```

Comment:
```suggestion
      .where(categories: { title: name })
```
через знак вопроса обещали совсем убрать возможность подстановки аргумента
оставят подстановку по ключу
но лучше всего, где это возможно, использовать "хеш-синтаксис" (см. выше ^^) - обычно это лаконичней и легче править